lib/app/cli.ts

```
@@ -0,0 +1,249 @@
+// Copyright (c) 2025, Compiler Explorer Authors
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import child_process from 'node:child_process';
+import * as fs from 'node:fs';
+import path from 'node:path';
+import {Command} from 'commander';
+
+import {AppArguments} from '../app.interfaces.js';
+import {logger} from '../logger.js';
+import * as utils from '../utils.js';
+
+/**
+ * Parses a command line option into a number.
+ */
+export function parseNumberForOptions(value: string): number {
+    // Ensure string contains only digits (and optional leading minus sign)
+    if (!/^-?\d+$/.test(value)) {
+        throw new Error(`Invalid number: "${value}"`);
+    }
```

Any specific reason to keep this limitation vs just accept what parseInt does?

---

It seemed perfectly reasonable to not silently accept `--port 10240moo` as that's almost certainly a typo/mistake :)

---

That would cause the parseInt to return `NaN` which fails below though. I believe the difference would be supporting octal/hex/binary by accident.

---

(this was discovered by claude as it added tests and they unexpectedly did not fail :) )

---

if anything the `-?` should definitely not be part of this regex, negative port numbers should really not be possible. And then rename `parseNumberForOptions` to `parseNumberForPort`

---

done